polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -230,6 +231,23 @@ class LoopTools {
                    const LoopVersionCondition &versionCond) const;
 };
 
+//===----------------------------------------------------------------------===//
+// SYCL related functions
+//===----------------------------------------------------------------------===//
+
+/// Returns true if the given function has linkonce_odr linkage.
+bool isLinkonceODR(FunctionOpInterface);
+
+/// Return true if the given function is only called from GPU kernels.
```

Comment:
Return true if the function is called directly from a GPU kernel or from a function which is called directly from a GPU kernel.

Reply:
Added `getMaxDepthFromGPUKernel` utility instead, please check if that's clearer or not.